stockflux-securities-master/src/components/InputForm.js

```diff
@@ -94,22 +96,43 @@ const InputForm = ({ match }) => {
           });
         })
         .catch(err => {
-          if (err instanceof ValidationError) {
-            setMessages(err.messages);
-          } else {
-            setMessages([err.message]);
-          }
-          setFormState(stateEnum.error);
+          errorHandler(err);
         });
     }
   };
 
+  const onClickDelete = () => {
+    deleteSecurity(match.params.securityId)
+      .then(() => {
+        setState(InputFormState.success);
+        setMessages(["Security Successfully Deleted"]);
+        setRedirect(true);
+      })
+      .catch(err => {
+        errorHandler(err);
+      });
+  };
+
+  const renderRedirect = () => {
```

Comment:
Personally, I would remove this function altogether in favour of actually conditionally rendering the redirect. This function isn't *pure* since you're relying on a value (redirect) that exists in a scope outside of it. If this value changes at some point, then your function will return a different value. A better signature would be to do something like this:

    const renderRedirect = (redirect) => {

The redirect is then passed to the function and this makes it pure because it has no side effects and it will always act the same given the same input. I'd still remove it though in favour of conditionally rendering the redirect component where you need to.

Reply:
Yep, I've removed the function in favour of conditionally rendering